src/test/scala/com/bazaarvoice/sswf/WaitTimeTest.scala

```diff
@@ -0,0 +1,154 @@
+        Thread.sleep(2000)
+        waitStepInvocationCount -= 1
+        InProgress(Some(s"waiting!!"))
+      } else {
+        Success(None)
+      }
+    }
+}
+
+class WaitTimeTest extends FlatSpec {
+
+  val parser = new InputParser[String] {
+    override def serialize(input: String): String = input
+    override def deserialize(inputString: String): String = inputString
+  }
+  private val domain: String = "sswf-tests"
+  private val wf: String = "wait-test"
+  private val swf: AmazonSimpleWorkflowClient = new AmazonSimpleWorkflowClient()
+  private val logger: StdOutLogger = new StdOutLogger
+  val manager = new WorkflowManagement[String, WaitTimeSteps](domain, wf, "0.0", wf, swf, inputParser = parser, log = logger)
+  val definition = new WaitTimeTestWorkflowDef()
+  val actor = new StepActionWorker[String, WaitTimeSteps](domain, wf, swf, parser, definition, log = logger)
+  val decider = new StepDecisionWorker[String, WaitTimeSteps](domain, wf, swf, parser, definition, logger)
+
+  "Each Step" should " have 2 invocations " in {
+    manager.registerWorkflow()
+    val workflow: WorkflowExecution = manager.startWorkflow("B", "")
+    try {
+
+      assert(waitForStepResult(false).getResult === StepResult.serialize(Success(None)))
+
+      //wait step invocations begin...
+      assert(waitForStepResult(false).getResult === StepResult.serialize(InProgress(Some("waiting!!"))))
+
+      //2nd invocation for wait step
+      assert(waitForStepResult(true).getResult === StepResult.serialize(InProgress(Some("waiting!!"))))
+
+      //3rd invocation for wait step
+      assert(waitForStepResult(true).getResult === StepResult.serialize(InProgress(Some("waiting!!"))))
+
+      //4th invocation for wait step
+      assert(waitForStepResult(true).getResult === StepResult.serialize(InProgress(Some("waiting!!"))))
+
+      //5th invocation for wait step
+      assert(waitForStepResult(true).getResult === StepResult.serialize(Success(None)))
+
+      //final dummy step invocation
+      assert(waitForStepResult(false).getResult === StepResult.serialize(Success(None)))
+
+      val wfHistory = manager.describeExecution(workflow.getWorkflowId, workflow.getRunId)
+      assert(wfHistory.events.size() == 8)
+
+      val firstDummyStep = wfHistory.events.get(1)
+      assert(firstDummyStep.cumulativeActivityTime.toStandardSeconds.getSeconds == 2)
+      assert(firstDummyStep.invocations == 1)
+
+      val firstWaitStepInProgress = wfHistory.events.get(2)
+      assert(firstWaitStepInProgress.cumulativeActivityTime.toStandardSeconds.getSeconds == 2)
+      assert(firstWaitStepInProgress.invocations == 1)
+
+      val secondWaitStepInProgress = wfHistory.events.get(3)
+      assert(secondWaitStepInProgress.invocations == 2)
+
+      val firstWaitTime: Duration = new Duration(firstWaitStepInProgress.end.get, secondWaitStepInProgress.start)
+      assert(firstWaitTime.getStandardSeconds >= 2 && firstWaitTime.getStandardSeconds <= 3)
```

Comment:
maybe instrument the test driver so you can verify the actual requested waitRetry time, rather than checking the wall clock time.